src/AI/ModularAI.java

```diff
@@ -0,0 +1,181 @@
+package AI;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.PriorityQueue;
+
+import CommandingOfficers.Commander;
+import CommandingOfficers.CommanderAbility;
+import Engine.GameAction;
+import Engine.Utils;
+import Engine.XYCoord;
+import Engine.UnitActionLifecycles.CaptureLifecycle;
+import Terrain.GameMap;
+import Units.Unit;
+
+/**
+ * Base class for inversion of control of AI behaviors/composition
+ */
+public abstract class ModularAI implements AIController
+{
+  protected Commander myCo = null;
+
+  // Updated on turn init
+  protected ArrayList<XYCoord> unownedProperties;
+  // List of possible AI modes, in order of precedence
+  protected ArrayList<AIModule> aiPhases;
+  // Used to determine what the default priority for unit actions is
```

Comment:
A bit ambiguous/awkward. I'm assuming `unitOrderSetter` just defines the order in which units should act, but the comment seems to imply that action types are being ordered somehow.

Reply:
It sets the order that units are fed to the `aiPhases`, which will typically define the default order in which phases pick actions.

So yes, and also yes.

Reply:
That sounds more like yes, and basically no (except maybe really indirectly).
I'd rather make the comment unambiguous that we are setting the order in which units are considered for actions within each phase, and preclude the interpretation that we are setting action priorities in any real sense.
Yes, if we prioritize all ASSAULT units, we are implicitly prioritizing attack actions, but we are only explicitly ordering units, so that's what we should call out.

Reply:
Fair enough.